parser/parser_test.go

```diff
@@ -445,6 +605,14 @@ func TestOperatorPrecedenceParsing(t *testing.T) {
 			"add(a + b + c * d / f + g)",
 			"add((((a + b) + ((c * d) / f)) + g))",
 		},
+		{
+			"a * [1, 2, 3, 4][b * c] * d",
+			"((a * ([1,2,3,4][(b * c)])) * d)", // Why is whitespace removed in this output for arrays only?
+		},
+		{
+			"add(a * b[2], b[1], 2 * [1, 2][1])",
+			"add((a * (b[2])), (b[1]), (2 * ([1,2][1])))", // Why is whitespace removed in this output for arrays only?
```

Comment:
For whatever reason, I had to remove the spaces in the arrays for the output here. It didn't do that on any other Literal type, only arrays.